eZ/Publish/Core/IO/Exception/InvalidBinaryFileIdException.php

```diff
@@ -8,12 +8,22 @@
  */
 namespace eZ\Publish\Core\IO\Exception;
 
+use Exception;
 use eZ\Publish\Core\Base\Exceptions\InvalidArgumentValue;
 
 class InvalidBinaryFileIdException extends InvalidArgumentValue
 {
     public function __construct($id, $message = null)
     {
-        parent::__construct('BinaryFile::id', $id, 'BinaryFile');
+        $parameters = ['%id%' => $id];
+        $string = "Argument 'BinaryFile::id' is invalid: '%id%' is wrong value";
+        if ($message) {
+            $string .= ', %message%';
+            $parameters['%message%'] = $message;
+        }
+
+        $this->setMessageTemplate($string);
```

Comment:
We can't automate the extraction of such a string (static code and stuff). The parser I'm working on looks for the `setMessageTemplate()` method.

(note that this becomes obsolete if you do the class change suggested above)

Could you change the code to:

```php
if ($message) {
  $this->setMessageTemplate("Argument 'BinaryFile::id' is invalid: '%id%' is wrong value, %message%");
  $parameters['%message%'] = $message;
} else {
  $this->setMessageTemplate("Argument 'BinaryFile::id' is invalid: '%id%' is wrong value");
}
```

That way, both flavours will be made available for translation.